### PR TITLE
Add support for sending beacon data using GET request

### PR DESCRIPTION
**Pull request by jakewan**
CBSi beacons to NS1 are sent using GET requests. We could handle this by overriding Provider.sendBeacon, but this might be useful to others.

### CODE REVIEW

File: src/util/beacon.ts [hunk 1]
```diff
@@ -6,6 +6,8 @@ import { SendBeaconResult } from "../@types"
  */
 const hasBeaconSupport = "sendBeacon" in navigator
 
+export type BeaconMethod = "GET" | "POST"
```
Comment (phamann) [inline src/util/beacon.ts]: Not a blocker, but i'd consider using a [string enum](https://www.typescriptlang.org/docs/handbook/enums.html#string-enums) here instead, as we know the values are finite.
```suggestion
export enum BeaconMethod {
 Get = "GET",
 Post = "POST",
} 
```

File: src/util/beacon.ts [hunk 2]
```diff
@@ -26,7 +28,11 @@ const hasBeaconSupport = "sendBeacon" in navigator
 export default function beacon(
     url: string,
     data: string,
+    method: BeaconMethod,
 ): Promise<SendBeaconResult> {
+    if (method == "GET") {
```
Comment (phamann) [inline src/util/beacon.ts]: If we used an enum we could use the token here instead of the string:
```suggestion
    if (method == BeaconMethod.Get) {
```

**Review by phamann — APPROVED** [review]
LGTM - Looks like you'll need to rebase this branch off `main` once #9 is merged.

File: src/util/beacon.ts (hunk 2 above)
Comment (jakewan) [reply src/util/beacon.ts]: I suspect this makes life easier for the compiler to optimize file size. 👍

**Pull request merged**